packages/client/src/execution/preimage.ts

```diff
@@ -0,0 +1,21 @@
+import { DBKey, MetaDBManager } from '../util/metaDBManager'
+
+export class PreimagesManager extends MetaDBManager {
+  /**
+   * Returns the preimage for a given hashed key
+   * @param key the hashed key
+   * @returns the preimage of the hashed key
+   */
+  async getPreimage(key: Uint8Array): Promise<Uint8Array | null> {
+    return this.get(DBKey.Preimage, key)
```

Comment:
I now realize that this class "only" maps `get -> getPreimage` and `set -> setPreimage` :thinking: I'm not super sure if this is necessary, since it is clear what we are trying to get (`get(DBKey.Preimage, key)`). But not strong feelings.